### DISCUSSION TITLE
this is undefined with API composition

### DISCUSSION THREAD

**Discussion by mowabidev** [quasarframework/quasar#11156] [CLI - SSR mode] [answered]
Hi, I am using Quasar with API composition and I have an issue.

I just want to get parameters from the current route so within onMounted, I try to use `this.$route.params`, but I always get **"this is undefined"**. I get the same error every time I use this in the setup.

So there must be something I'm not getting. Can you please help me?

**Comment by hawkeye64** [comment, marked as answer]
```
import { useRoute } from 'vue-router'
```
Then in setup()

```
const $route = useRoute()
```

**Reply by mowabidev** [reply]
@hawkeye64 thank you very much
